.github/workflows/protocol-build-and-push-snapshot.yml

```diff
@@ -1,6 +1,9 @@
 name: Protocol Build & Push Image to AWS ECR
 
 on:  # yamllint disable-line rule:truthy
+  pull_request:
+    paths:
+      - 'protocol/**'
   push:
```

Comment:
Typically when I do this I don't do based on pull request. I add a rule to `push` with my branch name (add `td/setup-staging-1` to `branches`. I think `pull_request` configures the github workflow environment differently which might be causing our issue.

Reply:
For more context, copying over Roy's messsage from DM

> what happens is when you make a PR, github workflow will merge your branch into main and then run workflows. we then end up fetching the name of that temporary merge commit (unsure if you actually merge it if it will end up being the same commit hash)
but when you run workflow on a branch, it will just run it on the commit directly